### PR TITLE
New-EntraOauth2PermissionGrant GA and Beta

### PR DESCRIPTION
**Pull request by v-akarke**
…etaOauth2PermissionGrant

### CODE REVIEW

File: module/Entra/AdditionalFunctions/New-EntraOauth2PermissionGrant.ps1 [hunk 1]
```diff
@@ -0,0 +1,63 @@
+function New-EntraOauth2PermissionGrant {
+    [CmdletBinding(DefaultParameterSetName = 'InvokeByDynamicParameters')]
```
Comment (alexandair) [inline module/Entra/AdditionalFunctions/New-EntraOauth2PermissionGrant.ps1:2]: The name of a parameter set should ideally tell what is specific for the set of parameters and why it is different than other sets. (The parameter set name is visible when the user runs `Show-Command <command>`).

`InvokeByDynamicParameters` doesn't do that.

**Review by alexandair — CHANGES_REQUESTED** [review]

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+The `New-EntraBetaOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
+
+## Examples
+
+### Example 1: To grant authorization to impersonate all users
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
+New-EntraBetaOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'AllPrincipals' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All' -StartTime (Get-Date) -ExpiryTime (Get-Date).AddYears(1)
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:48]:
```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'AllPrincipals'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
    StartTime = Get-Date
    ExpiryTime = (Get-Date).AddYears(1)
}

New-EntraBetaOauth2PermissionGrant @params
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+ -StartTime <datetime> 
+ -ExpiryTime <datetime> 
+ [-PrincipalId <string>]
+ [-Scope <string>]  
+ [<CommonParameters>]
+```
+
+## Description
+
+The `New-EntraBetaOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:40]:
```suggestion
The `New-EntraBetaOauth2PermissionGrant` cmdlet creates a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the specified delegated permissions.
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+```
+
+This command Grant authorization to impersonate all users.
+
+### Example 2: To grant authorization to impersonate a specific user
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
+New-EntraBetaOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'Principal' -PrincipalId 'aaaaaaaa-bbbb-cccc-1111-222222222222' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All' -StartTime (Get-Date) -ExpiryTime (Get-Date).AddYears(1)
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:64]:
```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'Principal'
    PrincipalId = 'aaaaaaaa-bbbb-cccc-1111-222222222222'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
    StartTime = Get-Date
    ExpiryTime = (Get-Date).AddYears(1)
}

New-EntraBetaOauth2PermissionGrant @params
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1 AllPrincipals 28/06/2025 07:44:25             aaaaaaaa-bbbb-cccc-1111-222222222222 Dele...
+
+```
+
+This command Grant authorization to impersonate all users.
+
+### Example 2: To grant authorization to impersonate a specific user
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:63]:
```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+## Description
+
+The `New-EntraBetaOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
+
+## Examples
+
+### Example 1: To grant authorization to impersonate all users
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:47]:
```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -ConsentType
+
+Indicates if the client application is authorized to impersonate all users or only a specific user. "AllPrincipals" indicates authorization to impersonate all users, while "Principal" indicates authorization to impersonate a specific user. Consent on behalf of all users can be granted by an administrator. In some cases, non admin users are authorized to consent on behalf of themselves for certain delegated permissions. This is required and supports $filter (eq only).
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:94]:
```suggestion
Indicates whether the client application is authorized to impersonate all users or only a specific user.

- `AllPrincipals`: Authorizes the application to impersonate all users.
- `Principal`: Authorizes the application to impersonate a specific user.

An administrator can grant consent on behalf of all users. In some cases, non-admin users are authorized to consent on behalf of themselves for certain delegated permissions. This parameter is required and supports the $filter query (eq only).
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+Module Name: Microsoft.Graph.Entra.Beta
+online version:
+schema: 2.0.0
+---
+
+# New-EntraBetaOauth2PermissionGrant
+
+## Synopsis
+
+Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaOauth2PermissionGrant.md:22]:
```suggestion
Create a delegated permission grant using an oAuth2PermissionGrant object. This grant allows a client service principal to access a resource service principal on behalf of a signed-in user, with access restricted to the specified delegated permissions.
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+Module Name: Microsoft.Graph.Entra
+online version:
+schema: 2.0.0
+---
+
+# New-EntraOauth2PermissionGrant
+
+## Synopsis
+
+Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:22]:
```suggestion
Create a delegated permission grant using an oAuth2PermissionGrant object. This grant allows a client service principal to access a resource service principal on behalf of a signed-in user, with access restricted to the specified delegated permissions.
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -ConsentType
+
+Indicates if the client application is authorized to impersonate all users or only a specific user. "AllPrincipals" indicates authorization to impersonate all users, while "Principal" indicates authorization to impersonate a specific user. Consent on behalf of all users can be granted by an administrator. In some cases, non admin users are authorized to consent on behalf of themselves for certain delegated permissions. This is required and supports $filter (eq only).
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:92]:
```suggestion
Indicates whether the client application is authorized to impersonate all users or only a specific user.
- `AllPrincipals`: Authorizes the application to impersonate all users.
- `Principal`: Authorizes the application to impersonate a specific user.
An administrator can grant consent on behalf of all users. In some cases, non-admin users are authorized to consent on behalf of themselves for certain delegated permissions. This parameter is required and supports the $filter query (eq only).
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+ -ConsentType <string> 
+ -ResourceId <string> 
+ [-PrincipalId <string>]
+ [-Scope <string>]  
+ [<CommonParameters>]
+```
+
+## Description
+
+The `New-EntraOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:38]:
```suggestion
The `New-EntraOauth2PermissionGrant` cmdlet creates a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the specified delegated permissions.
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+## Examples
+
+### Example 1: To grant authorization to impersonate all users
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:45]:
```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+A1bC2dE3fH4iJ5kL6mN7oP8qR9sT0u a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1 AllPrincipals 28/06/2025 07:44:25             aaaaaaaa-bbbb-cccc-1111-222222222222 Dele...
+
+```
+
+This command Grant authorization to impersonate all users.
+
+### Example 2: To grant authorization to impersonate a specific user
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:61]:
```suggestion
Connect-Entra -Scopes 'DelegatedPermissionGrant.ReadWrite.All'
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+New-EntraOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'Principal' -PrincipalId 'aaaaaaaa-bbbb-cccc-1111-222222222222' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:62]:
```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'Principal'
    PrincipalId = 'aaaaaaaa-bbbb-cccc-1111-222222222222'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
}

New-EntraOauth2PermissionGrant @params
```

File: module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+The `New-EntraOauth2PermissionGrant` cmdlet Create a delegated permission grant using an oAuth2PermissionGrant object. This grant authorizes a client service principal to access a resource service principal on behalf of a signed-in user, with access limited to the granted delegated permissions.
+
+## Examples
+
+### Example 1: To grant authorization to impersonate all users
+
+```powershell
+Connect-Entra -Scopes 'Directory.ReadWrite.All'
+New-EntraOauth2PermissionGrant -ClientId '00001111-aaaa-2222-bbbb-3333cccc4444' -ConsentType 'AllPrincipals' -ResourceId 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1' -Scope 'DelegatedPermissionGrant.ReadWrite.All'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-v1.0/Microsoft.Graph.Entra/New-EntraOauth2PermissionGrant.md:46]:
```suggestion
$params = @{
    ClientId = '00001111-aaaa-2222-bbbb-3333cccc4444'
    ConsentType = 'AllPrincipals'
    ResourceId = 'a0a0a0a0-bbbb-cccc-dddd-e1e1e1e1e1e1'
    Scope = 'DelegatedPermissionGrant.ReadWrite.All'
}

New-EntraOauth2PermissionGrant @params
```

**Review by SteveMutungi254 — COMMENTED** [review]
Left some comments/suggestions.

**Review by SteveMutungi254 — APPROVED** [review]
LGTM.

**Review by KenitoInc — APPROVED** [review]

**Pull request merged**